java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -53,17 +50,23 @@ public SymbolicValue createSymbolicValue(Tree syntaxNode) {
       result = symbolMap.get(((VariableTree) syntaxNode).symbol());
     }
     if (result == null) {
-      result = new SymbolicValue.ObjectSymbolicValue(map.size()+ProgramState.EMPTY_STATE.constraints.size());
+      result = new SymbolicValue.ObjectSymbolicValue(map.size() + ProgramState.EMPTY_STATE.constraints.size());
       map.put(syntaxNode, result);
-      if(syntaxNode.is(Tree.Kind.IDENTIFIER)) {
+      if (syntaxNode.is(Tree.Kind.IDENTIFIER)) {
         symbolMap.put(((IdentifierTree) syntaxNode).symbol(), result);
-      } else if(syntaxNode.is(Tree.Kind.VARIABLE)) {
+      } else if (syntaxNode.is(Tree.Kind.VARIABLE)) {
         symbolMap.put(((VariableTree) syntaxNode).symbol(), result);
       }
     }
     return result;
   }
 
+  public SymbolicValue supersedeSymbolicValue(VariableTree variable) {
+    SymbolicValue result = new SymbolicValue.ObjectSymbolicValue(map.size() + ProgramState.EMPTY_STATE.constraints.size());
+    map.put(variable, result);
+    symbolMap.put(variable.symbol(), result);
+    return result;
+  }
 
   public SymbolicValue eval(ProgramState programState, Tree syntaxNode) {
     syntaxNode = skipTrivial(syntaxNode);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "syntaxNode". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)